src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1065,38 +1054,50 @@ def add_units(
         Each string in this list that matches a unit property will not be written to the NWBFile.
     skip_features: list of str
         Each string in this list that matches a spike feature will not be written to the NWBFile.
-    use_times: bool (optional, defaults to False)
-        If True, the times are saved to the nwb file using sorting.frame_to_time(). If False (default),
-        the sampling rate is used.
-    write_as: str (optional, defaults to 'units')
-        How to save the units table in the nwb file. Options:
-        - 'units' will save it to the official NWBFile.Units position; recommended only for the final form of the data.
-        - 'processing' will save it to the processing module to serve as a historical provenance for the official table.
-    units_name : str (optional, defaults to 'units')
-        The name of the units table. If write_as=='units', then units_name must also be 'units'.
-    units_description : str (optional)
-        Text description of the sorting table; recommended to included parameters of sorting method, curation, etc.
+    write_in_processing_module: bool (optional, defaults to False)
+        How to save the units table in the nwb file.
+        - True will save it to the processing module to serve as a historical provenance for the official table.
+        - False will save it to the official NWBFile.Units position; recommended only for the final form of the data.
+    units_table_name : str (optional, defaults to 'units')
+        The name of the units table. If write_as=='units', then units_table_name must also be 'units'.
+    unit_table_description : str (optional)
+        Text description of the units table; it is recommended to include information such as the sorting method,
+        curation steps, etc.
     write_waveforms : bool (optional, defaults to false)
         if True and sorting is a spikeextractors SortingExtractor object then waveforms are added to the units table
         after writing.
     """
+    if not isinstance(nwbfile, pynwb.NWBFile):
+        raise TypeError(f"nwbfile type should be an instance of pynwb.NWBFile but got {type(nwbfile)}")
+
     if isinstance(sorting, SortingExtractor):
-        sampling_frequency = sorting.get_sampling_frequency()
-        if sampling_frequency is None:
-            raise ValueError("Writing a SortingExtractor to an NWBFile requires a known sampling frequency!")
```

Comment:
What is the behavior when you convert a `SortingExtractor` that doesn't have a sampling frequency using `OldToNewSorting`? Does the resultant object have a default one or something?

Reply:
I actually don't know the answer to that. I am actually not sure about this piece of code that was like this in the previous implementation. My hunch is that this was required because the sampling frequency was required to scale the spike times to the appropriate units.

Reply:
Well, *why* sampling frequency is required for NWB is because it is the original `NwbSortingExtractor` required it (https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extractors/nwbextractors/nwbextractors.py#L1134, or technically an associated electrical series that it would then infer it from).

So that still hasn't technically changed for a roundtrip, unless the new `BaseSorting` has a case to handle it.

Reply:
Similar logic:
https://github.com/SpikeInterface/spikeinterface/blob/12b7863684b705e3d0ae8165bb3009143c70530c/spikeinterface/extractors/nwbextractors.py#L279-L294

Reply:
But if that's the case, the assertion message seems misleading. We needed it for the round-trip test not because it was necessary to write the table.

Reply:
Hmm, it must have been for the `use_times`, then.

Reply:
Should we remove it then? `spikeinterface` will throw an assertion if they don't have the times and our assertion is misleading.